### ISSUE TITLE
Add syn_rename or similar?

### ISSUE THREAD

**Issue by kaberett** [dreamwidth/dw-free#1833]
http://www.dreamwidth.org/support/see_request?id=33755 has just shown up requesting that a feed account be renamed. There IS a way for me to do this currently, but it relies on exploiting a bug with the feed creator (that doesn't spot that two URLs identical apart from a trailing slash are... actually identical), or straight-up using an incorrect URL, to create a feed with the correct name, which I can then use syn_merge on, which seems a bit convoluted.

I get this kind of support request infrequently but not never; I'd be willing to do the dev work on this if given the go-ahead. But equally "special-casing renames of feed accounts" is not, um, *obviously* a correct thing to do, so.

**Comment by rahaeli** [comment]
Hm. I can do it (so you can xfer them to Accounts for now), but it involves logging into the admin server and using one of the command-line tools. I'd be fine with a console command to simplify it! Question is, do we want to make it syn_rename (and hang it off the synmerge privs), or do we want to make a console command that will let people rename any account and have a sub-priv for syn-only renames ... hrm. I'd say let's go with "make syn_rename and have it only work on feed accounts", since just about anything else can be handled by the "issue a free rename token and/or log into the admin box and command-line-rename something".

**Comment by kaberett** [comment]
\o/ Claim!

**Comment by kaberett** [comment]
Huh. https://github.com/dreamwidth/dw-free/blob/develop/cgi-bin/DW/User/Rename.pm line 124 claims renaming only works on community or personal journals; unless anyone yells otherwise I'm going to be reworking DW::User::Rename to also handle feed accounts in addition to writing up the console command ~frontend.

**Comment by kareila** [comment]
I... think that's only because we don't want to support people buying rename tokens for feed accounts? You may not need to use that module to do what you need for your console command.

**Comment by kaberett** [comment]
I *suspect*, from the skimming I've done so far, that the tidiest/safest way to set up syn_rename is to specialcase feed renaming in DW::User::Rename to check that the new name is acceptable (i.e. it meets all other criteria for usernames, *and* the last five characters are "_feed"). 

In principle this isn't necessary because you'll need syn_edit to do these renames anyway so *theoretically* the risk for human error & screw-ups is small, but it still seems to me that it would be better to check that the new username is appropriate via DW::User::Rename than to duplicate the code?

**Comment by kareila** [comment]
Agreed on not duplicating code; if that's the case, you may need to refactor things to make sure the universally applicable bits are separate from the token/shop related logic, and to make sure you're not enabling any unforeseen consequences from adding new capabilities to code that was originally written to be used elsewhere under a different set of assumptions about allowed behavior. If you've never written tests before, this might be a good opportunity for that... ;)

**Comment by rahaeli** [comment]
Hm. On-site renames are complicated enough that I wonder if this might be a place to just go ahead and do the code duplication, especially since they're two separate cases -- we don't need things like redirects, etc, for feed accounts. I would probably just look to bin/rename_user.pl instead.

**Comment by kaberett** [comment]
Okay right yes that is obviously sensible! Thank youuuuuu I will do that.

**Comment by kaberett** [comment]
... okay *now* I'm trying to convince myself that actually what I want to do is set up a --rename option for syn_merge (or syn_edit I guess) and duplicate the code THERE rather than in DW::User::Rename, which is clearly a terrible idea in terms of code being unfindable and is therefore PROBABLY an indication I should come back to this after some sleep.
